### PR TITLE
[CI] Created new CI job pipeline for Navi4x architecture

### PR DESCRIPTION
**Pull request by stefankoncarevic**
This PR adds a new CI pipeline specifically for the Navi4x architecture. Most configurations and functionalities have been adapted from the main Jenkins file, with adjustments for Navi4x. Created stages for setting system properties, environment setup, and executing PR CI tests and E2E tests. 
Please review the Jenkins file and provide any suggestions for further improving the testing process.

### CODE REVIEW

**Comment by manupak** [comment]
Thanks @stefankoncarevic . To me, this feels much cleaner.
Would you be able to set up a cron job using this branch + jenkinsfile ?

**Comment by stefankoncarevic** [comment]
That's right, I take this branch and the script I made and run CI, so I debug if there are any errors.

File: mlir/utils/jenkins/Jenkinsfile.navi4x [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    return "--device=/dev/kfd --device=/dev/dri --group-add video --group-add render -v /etc/passwd:/etc/passwd:ro -v /etc/group:/etc/group:ro"
+}
+
+String dockerImage() {
+    // If this is being changed please change Dockerfile.migraphx-ci's base image as well
+    return 'rocm/mlir:rocm6.2-latest'
+}
+
+void setHeartbeat() {
+    script {
```
Comment (krzysz00) [inline mlir/utils/jenkins/Jenkinsfile.navi4x]: Why all these `script {}` blocks? I'm pretty sure we don't need them?

Comment (stefankoncarevic) [reply mlir/utils/jenkins/Jenkinsfile.navi4x]: Not sure. Copied from our nightly. I assumed it was needed or at least wouldn't hurt. I will remove it.

**Comment by pcf000** [comment]
Looks good to me.  I know there's a way to share common code among Jenkinsfiles;  with three now, we should use it, but that's outside the scope here.

**Review by pcf000 — APPROVED** [review]

**Pull request merged**